pkg/kubelet/kubelet.go

```diff
@@ -1629,6 +1665,30 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 		return errOuter
 	}
 
+	// Create Cgroups for the pod and apply resource parameters to them if cgroup-per-qos flag is enabled.
+	pcm := kl.containerManager.NewPodContainerManager()
+	// If pod has already been terminated then we need not create or update the pod's cgroup
+	if !kl.podIsTerminated(pod) {
```

Comment:
Run once pods will fail in this case. If the pod was a run once pod and it was running and the kubelet is restarted with the flag enabled the pod will fail after killpod is called.

Sorry if I might have forgotten, but I thought this is what we agreed upon.